Adds info about watchtowers to the security page and glossary (#798)
BitcoinDesign/Guide: guide/daily-spending-wallet/security.md

---

## Preventing theft when offline

> Mobile wallets may frequently be offline, either due to a poor connection or because the application is not running. To improve battery live, mobile operating systems tend to prevent applications the user is not actively using from running code in the background. This can prevent mobile wallets from being reliable monitors of its payment channels. Malicious actors who share channels with the user may take advantage of these moments to try to steal funds by closing channels with incorrect data.

Reviewer: ```suggestion
Lightning wallets need to be online to monitor their payment channels to prevent their counter parties from attempting to steal their bitcoin. Mobile wallets however are frequently offline. This is usually due to a poor connection or the user not having the wallet open on their device. 

Most major mobile operating systems do not allow apps open in the background to run tasks, such as monitoring a lightning node. This is to improve battery life as well as prevent apps acting maliciously in the background. 

This prevents mobile lightning wallets from being reliable monitors of its payment channels. Malicious actors could take advantage of this and attempt to steal the mobile wallet users bitcoin by closing channels with incorrect data.
```

Stated the problem from the start and some other copy / readability changes.

> A simple technique to reduce this risk is to notify the user when the application has been offline for an extended period of time. [Forced channel closes](https://wiki.ion.radar.tech/tech/channels/channel-closing#force-close) have an agreed upon delay built-in. Notifications should be sent early enough so that users have time to appropriately respond.

Reviewer: ```suggestion
A technique to reduce this risk is to notify the user when the application has been offline for an extended period of time to open the app so their channels can be checked. [Forced channel closes](https://wiki.ion.radar.tech/tech/channels/channel-closing#force-close) have an agreed upon delay built-in. Notifications should be sent earlier than this delay so that users have time to appropriately respond.
```

> [Watchtowers](https://wiki.ion.radar.tech/tech/research/watchtowers) address this problem by continuously monitoring wallets and securely acting on the owners behalf when a problematic situation occurs. Watchtowers services should be provided by a different party than the wallet and payment channels, to reduce the risk of collusion.

Reviewer: ```suggestion
[Watchtowers](https://wiki.ion.radar.tech/tech/research/watchtowers) can also be used. These are third-parties that continuously monitor wallets and punish bad actors who attempt to cheat the wallet they are monitoring. Watchtowers should be provided by a different party than the wallet and payment channels, to reduce the risk of collusion.
```

> {% include image-gallery.html pages = page.images_watchtower %}

Reviewer: The contrast of the grey used on the iPhone frame and the background is not very good. It does not meet the [WCAG contrast guidelines](https://usecontrast.com/guide)  - we should just remove that grey background ;) https://github.com/BitcoinDesign/Guide/issues/674. There is is a lot frames with this issue fyi - can make another issue for this.

Author: WCAG contrast guidelines are for text, not for subtle background colors. This is a separate issue.

Also I think WCAG 2.0 needs to be interpreted as a guideline. My main problem with them is that text sizes 18 and up is treated the same. But, for example, 64pt text is so much larger that lower contrast still works. If we want a very-high-contrast version we should implement [prefers-contrast](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-contrast), a CSS media query that allows users/browsers to instruct websites to beef up the contrast. It has [80% support](https://caniuse.com/?search=prefers-contrast).

---

## Preventing theft when offline

> Lightning wallets need to be online to monitor their payment channels to prevent their counter parties from attempting to steal their bitcoin. Mobile wallets however are frequently offline. This is usually due to a poor connection or the user not having the wallet open on their device. 

Reviewer 2: ```suggestion
Lightning wallets need to be online to monitor their payment channels to prevent their counterparties from attempting to steal their bitcoin. Mobile wallets, however, are frequently offline, usually due to a poor connection or the user not having the wallet open on their device.
```

Author: [Hemingway](https://hemingwayapp.com) complains about the second, combined sentence being hard to read. It also complains about the first sentence being very hard to read. Wonder if we should do more general revision based on it.

Just a comment, accepting your change anyways.

Reviewer 2: I'm surprised Hemingway would be that opinionated about the changes. "Counterparties" as a single word is just a spelling correction. The commas surrounding "however" are a grammatical necessity.

Perhaps it's upset about replacing "This is" with a comma? I don't have strong feelings about that one. Sometimes too much "this, this, this" can make it difficult to know what "this" refers to, but hopefully it's clear in this context. Maybe Hemingway will be happy if you bring the"this" back?

> Most major mobile operating systems do not allow apps open in the background to run tasks, such as monitoring a lightning node. This is to improve battery life as well as prevent apps acting maliciously in the background. 

Reviewer 2: ```suggestion
Most major mobile operating systems do not allow apps open in the background to run tasks, such as monitoring a lightning node. This restriction is intended to improve battery life and prevent apps from acting maliciously in the background. 
```

> This prevents mobile lightning wallets from being reliable monitors of its payment channels. Malicious actors could take advantage of this and attempt to steal the mobile wallet users bitcoin by closing channels with incorrect data.

Reviewer 2: ```suggestion
However, this prevents mobile lightning wallets from being reliable monitors of their payment channels. Malicious actors could take advantage of this and attempt to steal a mobile wallet user's bitcoin by closing channels with incorrect data.
```

> A technique to reduce this risk is to notify the user when the application has been offline for an extended period of time to open the app so their channels can be checked. [Forced channel closes](https://wiki.ion.radar.tech/tech/channels/channel-closing#force-close) have an agreed upon delay built-in. Notifications should be sent earlier than this delay so that users have time to appropriately respond.

Reviewer 2: ```suggestion
One technique to reduce this risk is to notify the user when the application has been offline for an extended period. This way, they can open the app and thereby enable it to check their channels. Forced channel closes have a built-in, agreed-upon delay. Send notifications earlier than this delay so that users have time to respond appropriately.
```

> [Watchtowers](https://wiki.ion.radar.tech/tech/research/watchtowers) can also be used. These are third-parties that continuously monitor wallets and punish bad actors who attempt to cheat the wallet they are monitoring. Watchtowers should be provided by a different party than the wallet and payment channels, to reduce the risk of collusion.

Reviewer 2: ```suggestion
Another technique is to use [watchtowers](https://wiki.ion.radar.tech/tech/research/watchtowers). These are third parties that continuously monitor wallets and punish bad actors who attempt to cheat the wallet they are monitoring. Watchtowers should be provided by a different party than the wallet and payment channels to reduce the risk of collusion.
```
